src/Umbraco.Core/Routing/DefaultUrlAssembler.cs

```diff
@@ -0,0 +1,49 @@
+using Umbraco.Cms.Core.Models.PublishedContent;
+
+namespace Umbraco.Cms.Core.Routing;
+
+public class DefaultUrlAssembler : IUrlAssembler
+{
+    private readonly UriUtility _uriUtility;
+
+    public DefaultUrlAssembler(UriUtility uriUtility)
+    {
+        _uriUtility = uriUtility;
+    }
+
+    public Uri AssembleUrl(string path, Uri current, UrlMode mode)
+    {
+        if (string.IsNullOrWhiteSpace(path))
+        {
+            throw new ArgumentException($"{nameof(path)} cannot be null or whitespace", nameof(path));
+        }
+
+        // the stored path is absolute so we just return it as is
+        if (Uri.IsWellFormedUriString(path, UriKind.Absolute))
+        {
+            return new Uri(path);
+        }
+
+        Uri uri;
+
+        if (current == null)
+        {
+            mode = UrlMode.Relative; // best we can do
+        }
```

Comment:
We have specified in the method signature that this isn't nullable. 
I think we should either:

1. Make `Uri current` nullable, and then we can have this bit of code.
2. Fail fast. Throw an exception if this is null `ArgumentNullException.ThrowIfNull(current);`